File: 명관/20230607/성적 통계 5800.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int k, n, c;
int cmp(const int &a, const int &b)
{
    return a > b;
}
int main() {
    cin >> k;
    while (k--)
    {
        int lg = 0;
        cin >> n;
        int *a = new int[n];
        for (int i = 0; i < n; i++)
            cin >> a[i];
        sort(a,a+n,cmp);
        for (int i = 0; i < n - 1; i++)
            if (lg < a[i] - a[i+1])
                lg = a[i] - a[i+1];
        cout << "Class " << ++c<<'\n';
        cout << "Max " << a[0] << ", Min " << a[n-1] << ", Largest gap " << lg <<'\n';
    }
}

File: 명관/20230607/성적 통계 5800.java
import java.util.*;
import java.io.*;
 
public class Main {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
				StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();
        int k = Integer.parseInt(st.nextToken());
        for (int i = 0; i < k; i++) {
						st = new StringTokenizer(br.readLine());
            int N = Integer.parseInt(st.nextToken());
 
            Integer arr[] = new Integer[N];
 
            for (int j = 0; j < N; j++) {
                arr[j] = Integer.parseInt(st.nextToken());
            }
 
            Arrays.sort(arr, Collections.reverseOrder());
 
            int max = arr[0];
            int min = arr[N - 1];
 
            int sub = Integer.MIN_VALUE;
 
            for (int j = 0; j < arr.length - 1; j++) {
                sub = Math.max(arr[j] - arr[j + 1], sub);
            }
 
            sb.append("Class " + (i + 1) + "\n");
            sb.append("Max " + max + ", " + "Min " + min + ", " + "Largest gap " + sub + "\n");
        }
 
        System.out.println(sb);
    }
}